code/obj/item/swingsign.dm

```diff
@@ -0,0 +1,182 @@
+#define CHARACTERLIMIT 350
+
+//1. /obj/swingsign
+//	Inherited procs
+//	tgui procs
+//	procs
+//2. /obj/item/swingsignfolded
+
+TYPEINFO(/obj/swingsign)//No idea what TYPEINFO is, I just know it lets me disable the brown overcoat when crafting signs
+	mat_appearances_to_ignore = list("wood")
+/obj/swingsign
+	name = "swing sign"
+	desc = "A foldable sign for writing annoucements or advertisements."
+	icon = 'icons/obj/furniture/swingsign.dmi'
+	icon_state = "blank"
+	throwforce = 10
+	density = 1
+	anchored = UNANCHORED
+	/// Stored message
+	var/message = ""
+	var/defaultdesc = "A foldable sign for writing annoucements or advertisements."
+	/// There's a /div in SetMessage. If changing the preamble remember to respect it.
+	var/descpreamble = "It says:<br><div style='text-align:center'>"
+	var/secured = FALSE
+	/// Damage when thrown into a swing sign
+	var/maxcrashdamage = 5
+	/// Max length of the message
+	var/maxmessagerows = 10
+	/// Max width of the message
+	var/maxmessagecols = 40
+
+	proc/setmessage(var/newmessage)
+		message = newmessage
+		if(message == "")
+			desc = defaultdesc
+			icon_state = "blank"
+		else
+			desc = descpreamble + message + "</div>"
+			icon_state = "written"
+
+	proc/fold()
+		var/obj/item/swingsignfolded/C = new/obj/item/swingsignfolded(src.loc)
+
+		if (src.material)
+			C.setMaterial(src.material)
+		if (src.icon_state)
+			C.icon_state = "folded"
+		if (src.message)
+			C.message = src.message
+
+		qdel(src)
+
+	attack_hand(mob/user)
+		if (!isliving(user)) return
+		if(!secured)
+			user.visible_message("<b>[user.name] folds [src].</b>")
+			playsound(src, 'sound/impact_sounds/Clock_slap.ogg',50,1)
+			fold()
+		else
+			boutput(user, "<span alert='notice'>Sign is too tightly secured to fold!</span>")
+		return
+
+	attackby(obj/item/W, mob/user)
+		if(istype(W, /obj/item/pen))
+			ui_interact(user)
+			return
+		else if (isscrewingtool(W))
+			if(secured)
+				boutput(user, "<span class='notice'>You unsecure the swing sign.</span>")
+				anchored = UNANCHORED
+			else
+				boutput(user, "<span class='notice'>You secure the swing sign.</span>")
+				anchored = ANCHORED
+			playsound(src, 'sound/items/Screwdriver.ogg', 50, 1)
+			secured = !secured
+			return
+		..()
+
+	hitby(atom/movable/M, datum/thrown_thing/thr)
+		if (isliving(M) && M.throwing)
+			var/mob/living/L = M
+			var/area/T = get_area(src)
+			if(T?.sanctuary)
+				return
+			random_brute_damage(M, rand(1,maxcrashdamage),1)
+			L.force_laydown_standup()
+			L.changeStatus("stunned", 1 SECONDS)
```

Comment:
should be using `do_disorient`

Reply:
```suggestion
			L.do_disorient(25, 1 SECOND, stack_stuns = FALSE)
```
personal opinion. Adds on stamina damage, changes from stunned to weakened (aka knocked down), prevents stun stacking

Reply:
Tweaked a couple of parameters and kept force_laydown_standup. For some reason mobs thrown at the sign do not trip without it, even though it's already in the do_disorient code.